code/optimize/src/test/java/com/adobe/marketing/mobile/optimize/OfferTests.java

```diff
@@ -318,42 +320,44 @@ public void testFromEventData_invalidOfferNoContent() throws Exception {
     }
 
     @Test
-    public void testFromEventData_invalidOfferNoFormat() throws Exception {
+    public void testFromEventData_invalidOfferNoItemData() throws Exception {
         Map<String, Object> offerData =
                 new ObjectMapper()
                         .readValue(
                                 getClass()
                                         .getClassLoader()
-                                        .getResource("json/OFFER_INVALID_MISSING_FORMAT.json"),
+                                        .getResource("json/OFFER_INVALID_MISSING_ITEM_DATA.json"),
                                 HashMap.class);
         final Offer offer = Offer.fromEventData(offerData);
         Assert.assertNull(offer);
     }
 
     @Test
-    public void testFromEventData_invalidOfferNoItemData() throws Exception {
+    public void testFromEventData_invalidOfferIdMismatch() throws Exception {
         Map<String, Object> offerData =
                 new ObjectMapper()
                         .readValue(
                                 getClass()
                                         .getClassLoader()
-                                        .getResource("json/OFFER_INVALID_MISSING_ITEM_DATA.json"),
+                                        .getResource("json/OFFER_INVALID_ID_MISMATCH.json"),
                                 HashMap.class);
         final Offer offer = Offer.fromEventData(offerData);
         Assert.assertNull(offer);
     }
 
-    @Test
-    public void testFromEventData_invalidOfferIdMismatch() throws Exception {
+    @Test(expected = DataReaderException.class)
+    public void testFromEventData_invalidOfferFormatTypeIsNotString() throws Exception {
         Map<String, Object> offerData =
                 new ObjectMapper()
                         .readValue(
                                 getClass()
                                         .getClassLoader()
-                                        .getResource("json/OFFER_INVALID_ID_MISMATCH.json"),
+                                        .getResource("json/OFFER_INVALID_FORMAT_TYPE.json"),
                                 HashMap.class);
-        final Offer offer = Offer.fromEventData(offerData);
-        Assert.assertNull(offer);
+        final Map<String, Object> offerDataMap =
+                DataReader.getTypedMap(
+                        Object.class, offerData, OptimizeConstants.JsonKeys.PAYLOAD_ITEM_DATA);
+        DataReader.getString(offerDataMap, OptimizeConstants.JsonKeys.PAYLOAD_ITEM_DATA_FORMAT);
```

Comment:
What are we testing here? We need to check the behavior when a map with non string format is passed to `Offer.fromEventData()`. It should return null since the `DataReaderException` is already caught. Could you please fix this test?

Reply:
fixed @spoorthipujariadobe